onecore_maintenance_extension/models/maintenance.py

```diff
@@ -192,12 +192,13 @@ def update_form_options(self, search_by_number, search_type):
                     for tenant in lease['tenants']:
                         # Check if a tenant with the same contact_code already exists
                         existing_tenant = self.env['maintenance.tenant.option'].search([('contact_code', '=', tenant['contactCode'])], limit=1)
-                            
+
                         if not existing_tenant:
+                            name = (tenant.get('firstName') and tenant.get('lastName') and (tenant['firstName'] + " " + tenant['lastName'])) or tenant.get('fullName', '')
```

Comment:
I think this would be a bit more readable if it was broken up into an if/else.

Reply:
I agree! Good point!